### ISSUE TITLE
Unable to use nightly host tools when using bzlmod

### ISSUE THREAD

**Issue by csmulhern** [bazelbuild/rules_rust#2441]
The tag class definition for rust.host_tools includes `version`, but not `iso_date`.

https://github.com/bazelbuild/rules_rust/blob/fe610dab73c3c61560a547bc661177dd5115fed7/rust/extensions.bzl#L126-L129

This leads to the following error when trying to use a nightly version of the rust host tools (which is e.g. required to enable bindeps in crate universe).

```bash
ERROR: <builtin>: fetching rust_toolchain_tools_repository rule //:rules_rust~0.38.0~rust~rust_host_tools: Traceback (most recent call last):
	File "/private/var/tmp/_bazel_cameron/f5a6fe93ded5fb771eddb255ace4ff75/external/rules_rust~0.38.0/rust/repositories.bzl", line 295, column 24, in _rust_toolchain_tools_repository_impl
		check_version_valid(ctx.attr.version, ctx.attr.iso_date)
	File "/private/var/tmp/_bazel_cameron/f5a6fe93ded5fb771eddb255ace4ff75/external/rules_rust~0.38.0/rust/private/repository_utils.bzl", line 622, column 13, in check_version_valid
		fail("{param_prefix}iso_date must be specified if version is 'beta' or 'nightly'".format(param_prefix = param_prefix))
Error in fail: iso_date must be specified if version is 'beta' or 'nightly'
```

**Comment by cameron-martin** [comment]
Possibly fixed by #2545?

**Comment by csmulhern** [comment]
I think so. Thanks!
